tests/models/test_default_evaluator.py

```diff
@@ -636,25 +646,110 @@ def example_custom_metric_with_artifacts(given_df, given_metrics):
     assert res_artifacts_2["example_dictionary_artifact"] == {"a": 1, "b": 2}
 
 
-def test_custom_metric(binary_logistic_regressor_model_uri, breast_cancer_dataset):
-    def example_custom_metric(eval_df, given_metrics):
-        return {
-            "true_count": given_metrics["true_negatives"] + given_metrics["true_positives"],
-            "positive_count": np.sum(eval_df["prediction"]),
-        }
+def test_infer_artifact_type_and_ext_raise_exception_for_non_file_artifact_path(tmp_path):
```

Comment:
Do we need to be testing these internal utils, or can we rely on integration tests of public API behavior?

Reply:
I feel like there's no harm in having more verbose tests, I think this makes it easier to quickly spot behavior changes in case of future changes to custom metrics. Also, it seems like there's existing tests that [test for internal functions](https://github.com/mlflow/mlflow/blob/91ed65c0a756758f3bb7cfa4d41071d5d1393962/tests/models/test_default_evaluator.py#L281-L505)

Reply:
I don't see what these tests add by themselves. I'd rather we focus on the behavior of the user-facing `mlflow.evaluate()` API.